File: todo/frontend/src/types/todo.ts
// 对应后端
export interface TodoBase{
    title: string;
    details?: string;
    completed: boolean;
}

export interface TodoCreate extends TodoBase{}

export interface TodoInfo extends TodoBase {
    id: string; // UUID 以字符串形式表示
    created_at: string;  // 日期时间以 ISO 字符串形式表示
}

File: todo/backend/app/models/todo.py
from sqlalchemy import Column, String, Boolean, DateTime
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime, timezone
from app.core.database import Base

class Todo(Base):
    __tablename__ = "todos"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title = Column(String(100), nullable=False)
    details = Column(String(200), nullable=True)
    completed = Column(Boolean, nullable=False, default=False)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))